src/theory/logic_info.cpp

```diff
@@ -385,21 +390,45 @@ void LogicInfo::setLogicString(std::string logicString)
     p += 3;
   } else if(!strcmp(p, "QF_ALL")) {
     // the "all theories included" logic, no quantifiers
-    enableEverything();
+    if (d_higherOrder)
```

Comment:
Can you explain why `d_higherOrder` is needed to be checked here?  Also, enableEverything should be pulled from each branch; then delete the else branch.

Reply:
The check is necessary because `enableEverything()` invokes the default constructor of `LogicInfo`, which sets `d_higherOrder` to false, since the "everything" case does not include higher-order. This is why the test is made s.t. if we are in the higher-order mode, after we enable everything (and thus set ho to false) we reset higher order to true.

Reply:
Ok, just refactor the branches then.

Reply:
Also can't do that. I need to first check `d_higherOrder` before enabling everything and I need to do `enableHigherOrder` after `enableEverything`. I added some documentation to try to clarify that.

Reply:
That does simplify it, thanks!